File: Algorithms/p283_Move_Zeroes/p283_Move_Zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        nums_len = len(nums)
        zs = 0
        for v in nums:
            if v==0:
                break
            else:
                zs += 1
        
        for i in range(zs+1,nums_len):
            if nums[i]!=0:
                nums[zs], nums[i] = nums[i], nums[zs]
                zs += 1

File: Algorithms/p283_Move_Zeroes/p283_Move_Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int len = nums.size();
        int zs;
        for(zs=0;zs<len;zs++){
            if(nums[zs]==0){
                break;
            }
        }
        for(int i=zs+1;i<len;i++){
            if(nums[i]!=0){
                int tmp = nums[zs];
                nums[zs] = nums[i];
                nums[i] = tmp;
                zs++;
            }
        }
    }
};
